plugins/cilium-cni/lib/deletion_queue.go

```diff
@@ -164,8 +165,35 @@ func (dc *DeletionFallbackClient) EndpointDeleteMany(req *models.EndpointBatchDe
 // indicating the caller to attempt fallback logic.
 func (dc *DeletionFallbackClient) deleteEndpointsBatch(req *models.EndpointBatchDeleteRequest) (bool, error) {
 	if err := dc.tryConnect(); err != nil {
-		// Failed to setup cilium client.
-		return true, err
+		// Check if agent is starting up. If so, it is about to handle the
+		// deletion queue, thus we should avoid taking the lock in order to
+		// reduce contention.
+		// Instead, we wait for it to complete its bootstrap and retry to
+		// connect again later.
+		var dirExists, socketExists bool
+		if _, err := os.Stat(filepath.Dir(client.DefaultSockPath())); errors.Is(err, fs.ErrNotExist) {
```

Comment:
AFAICT this is true if the directory doesn't exist. Shouldn't we instead check the opposite?

```suggestion
		if _, err := os.Stat(filepath.Dir(client.DefaultSockPath())); !errors.Is(err, fs.ErrNotExist) {
```